### PR TITLE
Constrain calculator computation type

### PR DESCRIPTION
**Pull request by dkubb**

### CODE REVIEW

File: core/app/models/spree/calculator.rb [hunk 1]
```diff
@@ -13,12 +15,17 @@ class Calculator < Spree::Base
     def compute(computable)
       # Spree::LineItem -> :compute_line_item
       computable_name = computable.class.name.demodulize.underscore
-      method = "compute_#{computable_name}".to_sym
-      calculator_class = self.class
-      if respond_to?(method)
-        self.send(method, computable)
+      method          = :"compute_#{computable_name}"
+      amount          = __send__(method, computable) || DEFAULT_AMOUNT
```
Comment (mbj) [inline core/app/models/spree/calculator.rb]: Do we have any calculator that would legitimately return nil so this default would be taken?

File: core/app/models/spree/calculator/flat_percent_item_total.rb [hunk 2]
```diff
@@ -1,17 +1,22 @@
 require_dependency 'spree/calculator'
 
 module Spree
-  class Calculator::FlatPercentItemTotal < Calculator
-    preference :flat_percent, :decimal, default: 0
+  class Calculator
+    class FlatPercentItemTotal < Calculator
```
Comment (mbj) [inline core/app/models/spree/calculator/flat_percent_item_total.rb]: Do we have a legitimate (non spec) use case for this calculator? If not we may be able to remove it and use a subclass from specs?

File: core/app/models/spree/calculator/percent_on_line_item.rb [hunk 3]
```diff
@@ -1,15 +1,20 @@
+require_dependency 'spree/calculator'
+
 module Spree
   class Calculator
     class PercentOnLineItem < Calculator
```
Comment (mbj) [inline core/app/models/spree/calculator/percent_on_line_item.rb]: Instance of the question in: https://github.com/MountainRoseHerbs/spree/pull/597/commits/63c0674c751478b25fda91160cb4d22df4bb52ef#r92711132

File: core/app/models/spree/calculator/shipping/flat_rate.rb [hunk 4]
```diff
@@ -3,15 +3,15 @@
 module Spree
   module Calculator::Shipping
     class FlatRate < ShippingCalculator
```
Comment (mbj) [inline core/app/models/spree/calculator/shipping/flat_rate.rb]: Instance of the question in: https://github.com/MountainRoseHerbs/spree/pull/597/commits/63c0674c751478b25fda91160cb4d22df4bb52ef#r92711132

File: core/app/models/spree/calculator.rb (hunk 1 above)
Comment (dkubb) [inline core/app/models/spree/calculator.rb]: @mbj ok this is removed. I kept the `DEFAULT_AMOUNT` constant since many of the subclasses use it.

**Comment by mbj** [comment]
@dkubb I suspect the only real calculator subclass of `Spree::Calculator` we use right now is the `ShippingCalculator`, I wounder if it makes sense to remove the other calculators, and if they are only used from specs: consoldidate these calcualtors into one and move them to a spec only subclass.

This would not be blocking for me, and can be done in anothre iteration. I mention it here as I prefer to remove (production) unused code than fixing it.

**Comment by dkubb** [comment]
> I mention it here as I prefer to remove (production) unused code than fixing it.

I'll keep the refactorings in place, but do separate commits for the removal of the two classes you flagged. This way if we ever restore those classes, at least they'll be in the best possible state.

I will look at the other calculators to see if they can be removed too.

**Pull request merged**